test/clojure-mode-sexp-test.el

```diff
@@ -0,0 +1,36 @@
+;;; clojure-mode-sexp-test.el --- Clojure Mode: sexp tests  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2015 Artur Malabarba <artur@endlessparentheses.com>
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Code:
+
+(require 'clojure-mode)
+(require 'ert)
+
+(ert-deftest test-sexp ()
+  (with-temp-buffer
+    (insert "^String #macro ^dynamic reverse")
```

Comment:
This probably needs a few more tests showing that it won't go back or forward too far (e.g. to a previous/next expression).